cleo/imaging/sensors.py

```diff
@@ -243,12 +243,103 @@ def params(self) -> dict:
+    k: Quantity = field(kw_only=True)
+    """the decay parameter of the model how quickly does normalized fluorescence decay as the brightness level increases past its maximum saturation level."""
+    x: float = field(kw_only=True)
+    """the brightness of the function for the hills exponential decay equation."""
+    n: float = field(kw_only=True)
+    """the steepness of the transition with higher values meaning a more sudden
+      more switch like response and a lower value meaning a smoother more gradual response function."""
+    ec50: float = field(kw_only=True)
+    """the value of the brightness where the response is halfway between the bottom and Fmax"""
+    #Multivariate log normal distribution of wavelength to fluorescence
+    baseline_w: float = field(kw_only=True)
+    """baseline of the log normal equation when the wavelength is 0."""
+    w: Quantity = field(kw_only=True)
+    """the wavelength of the function for the log normal equation."""
+    A1: Quantity = field(kw_only=True)
+    """the amplitude of the first log normal distribution highlighting the maximum height of that first log normal function."""
+    sigma1: Quantity = field(kw_only=True)
+    """the standard distribution or the width of the first log normal distribution highlighting how wide the curve is."""
+    A2: Quantity = field(kw_only=True)
+    """the amplitude of the second log normal distribution highlighting the maximum height of that second log normal function."""
+    sigma2: Quantity = field(kw_only=True)
+    """the standard distribution or the width of the second log normal distribution highlighting how wide the curve is."""
+    mu1: Quantity = field(kw_only=True)
+    """the mu1 is the mean of the distribution or the center of the first log normal distribution."""
+    mu2: Quantity = field(kw_only=True)
+    """the mu2 is the mean of the distribution or the center of the second log normal distribution."""
+    exc_factor: str = field(
+        default = """
+        dF_p = baseline + ((A) * x**n)/(ec50**n + x**n) * exp(-k * x) : 1
+        dF_w = baseline_w + ((A1 * (1/(w * sigma1 * (2 * pi)**0.5)) * exp(-(log(w) - mu1)**2/(2 * sigma1**2)))) + ((A2 * (1/(w * sigma2 * (2 * pi)**0.5)) * exp(-(log(w) - mu2)**2/(2 * sigma2**2)))) : 1
+        dFF = dF_p * dF_w : 1
+        """,
+        init=False,
+    )
+    cal_model: CalciumModel = field(kw_only=True)
+    bind_act_model: CalBindingActivationModel = field(kw_only=True)
+    exc_model: ExcitationModel = field(kw_only=True)
+    location: str = field(default="cytoplasm", init=True)
+    def __attrs_pre_init__(self):
+        if not hasattr(self, "name"):
+            self.name = "GECI"
+    def __attrs_post_init__(self):
+        super_post_init = getattr(super(), "_attrs_post_init_", None)
+        if callable(super_post_init):
+            super_post_init()
+
+    def fluorescence(self, GECI_str, brightness, wavelength):
```

Comment:
this looks like a testing sort of function? If the interface is clunky enough we need code like this to test it, we need to revise the interface. But I don't think so...look at `tests/test_sensors.py` and you'll see it should be as simple as instantiating it (e.g., `geci = gcamp7f()`, then injecting it.